### PR TITLE
Ali GÖK Module 7 Week 1

### PR DESCRIPTION
**Pull request by aligoknl**

### CODE REVIEW

File: week1/project/ecommerce/src/App.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+.category-item:hover {
+  background-color: #dcdceb;
+  cursor: pointer;
+}
+.category-item-selected {
+  background-color: #bdbdc1;
+}
+.products {
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.css:23]: Have you checked the responsiveness of this? All the items get crammed together on mobile! Consider using a more flexible layout structure or media queries. 🔴

File: week1/project/ecommerce/src/components/CategoryList.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import allCategories from "../fake-data/all-categories";
+import React from "react";
+
+function CategoryList({ setCategory, setActiveButton, activeButton }) {
+  const toggle = (index, item) => {
+    if (activeButton === index) {
```
Comment (crevulus) [inline week1/project/ecommerce/src/components/CategoryList.js:6]: Very nice filtering! Being able to deselect the category is not a requirement, so you have put in some good extra work here.

File: week1/project/ecommerce/src/components/CategoryList.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      setCategory("allCategories");
+      setActiveButton(null);
+    } else {
+      setCategory(item.substring(6));
```
Comment (crevulus) [inline week1/project/ecommerce/src/components/CategoryList.js:10]: Currently this toggle isn't working. If you select a category, all items disappear. All products show again when you deselect the category. Check what you're doing to mutate the data here - it conflicts with the data you're filtering through in `Products.js`. 🔴

File: week1/project/ecommerce/src/components/Product.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const Product = ({ item }) => {
```
Comment (crevulus) [inline week1/project/ecommerce/src/components/Product.js:1]: Nice destructuring

**Comment by crevulus** [comment]
Good work Ali. Clearly your extra work studying React has paid off, because this is a very well-functioning React app.

There are a few 🔴 comments. One of them is css-related, one of them is to do with data control (and ensuring data matches from data source to data source), and one of them is a README item you must have missed. None of them are major or causes for concern, but are nonetheless important for this week's homework!

Also, do you have a netlify link for me? Deploying the site was also a requirement this week.

File: week1/project/ecommerce/src/App.css (hunk 1 above)
Comment (aligoknl) [reply week1/project/ecommerce/src/App.css:30]: Thank you Chris. I have a just given a nice responsive mobile first design, I guess.

File: week1/project/ecommerce/src/components/CategoryList.js (hunk 2 above)
Comment (aligoknl) [reply week1/project/ecommerce/src/components/CategoryList.js:6]: Thanks.

File: week1/project/ecommerce/src/components/CategoryList.js (hunk 3 above)
Comment (aligoknl) [reply week1/project/ecommerce/src/components/CategoryList.js]: I was playing with the code and I did not notice there was a bug on the last one. I have changed and it is working very well now.

File: week1/project/ecommerce/src/components/Product.js (hunk 4 above)
Comment (aligoknl) [reply week1/project/ecommerce/src/components/Product.js:1]: Thanks.

File: week1/project/ecommerce/src/components/Products.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import allProducts from "../fake-data/all-products";
+import Product from "./Product";
+
+const Products = ({ category }) => {
+  const filterProducts = () => {
```
[the remark this answers is not in the document]
Comment (aligoknl) [reply week1/project/ecommerce/src/components/Products.js:5]: Yes, it is working perfectly, indeed. Thanks.

File: week1/project/ecommerce/src/components/CategoryList.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      setActiveButton(index);
+    }
+  };
+
+  return (
+    <div>
+      {allCategories.map((item, index) => {
+        return (
+          <button
+            key={index}
```
[the remark this answers is not in the document]
Comment (aligoknl) [reply week1/project/ecommerce/src/components/CategoryList.js:28]: Yes, I missed that part. But, now I added an extra class and customized it. Thanks.

File: week1/project/ecommerce/src/App.css [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  margin: 1rem;
+}
+
+.category-item {
+  padding: 12px;
+  margin: 2px;
+  background-color: #babaca;
+}
+
+.category-item:hover {
```
[the remark this answers is not in the document]
Comment (aligoknl) [reply week1/project/ecommerce/src/App.css:20]: Thank you.

File: week1/project/ecommerce/public/manifest.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  "short_name": "React App",
+  "name": "Create React App Sample",
+  "icons": [
+    {
+      "src": "favicon.ico",
+      "sizes": "64x64 32x32 24x24 16x16",
+      "type": "image/x-icon"
+    },
+    {
+      "src": "logo192.png",
```
[the remark this answers is not in the document]
Comment (aligoknl) [reply week1/project/ecommerce/public/manifest.json]: I cleaned it on the html.file. There is no error in the console now. Thank you.

**Comment by aligoknl** [comment]
> Good work Ali. Clearly your extra work studying React has paid off, because this is a very well-functioning React app.
> 
> There are a few 🔴 comments. One of them is css-related, one of them is to do with data control (and ensuring data matches from data source to data source), and one of them is a README item you must have missed. None of them are major or causes for concern, but are nonetheless important for this week's homework!
> 
> Also, do you have a netlify link for me? Deploying the site was also a requirement this week.

Thank you so much for your valuable time and your useful comments Chris.

**Review by crevulus — APPROVED** [review]

**Pull request closed without merging**